src/get-route-id-trip-id-shape-id-lookup.js

```diff
@@ -0,0 +1,31 @@
+const { createReadStream } = require('node:fs');
+const { resolve } = require("path");
+const { parse } = require('csv-parse');
+
+async function getRouteTripShapeLookup(unzippedGtfsPath) {
+  const tripsStream = createReadStream(resolve(unzippedGtfsPath, 'trips.txt'), {encoding: 'utf8'});
+  const parser = tripsStream.pipe(parse({columns: true}));
+
+  const table = {};
+
+  for await(const trip of parser) {
+    const routeId = trip['route_id'];
+    const tripId = trip['trip_id'];
+    const shapeId = trip['shape_id'];
+
+    if (routeId && tripId && shapeId) {
+      // Lazily init the first level of the dictionary
+      if (table[routeId] == null) {
+        table[routeId] = {};
+      }
+
+      table[routeId][tripId] = shapeId;
```

Comment:
This probably won't come up in practice, but using a JavaScript object as a dictionary like this can break if the keys (a route ID or trip ID) match the name of a property on the object prototype, like 'constructor', which is technically a valid route ID, trip ID, stop ID, etc.